src/connectedk8s/azext_connectedk8s/custom.py

```diff
@@ -137,6 +139,10 @@ def create_connectedk8s(cmd, client, resource_group_name, cluster_name, correlat
     if enable_private_link is not None or distribution_version is not None or azure_hybrid_benefit is not None:
         client = cf_connected_cluster_prev_2023_11_01(cmd.cli_ctx, None)
 
+    # Set preview client if the connection-type is provided. (TODO: To test whether overriding the client factory to 2024 will retain the 2023 private link feature as in the line above)
+    if connection_type is not None and connection_type == "gateway" or enable_workload_identity or enable_oidc_issuer:
```

Comment:
we should keep this api version as default unless specific scenarios require other versions which has custom changes